### PR TITLE
Entity Mapping Refactored (vol1)

### PR DESCRIPTION
**Pull request by f3l1x**
I've refactored whole entity mapping. Finally I am more happy with current solution.
More tests included. 

-----

Configuration for annotation and xml changed.

Before

```yaml
nettrine.orm.annotations:
  namespaces: [App\Model\Database]
  paths: [%appDir%/Model/Database]
```

After

```yaml
nettrine.orm.annotations:
  mapping:
    %appDir%/Model/Database: App\Model\Database
```

-----

Dropped TEntityMapping, introduced more powerfull EntityHelper.

Before

```php
class CategoryExtension extends CompilerExtension
{
  use TEntityMapping;
  public function beforeCompile(): void
  {
    $this->setEntityMappings([
      'Forum' => __DIR__ . '/../Entity',
    ]);
  }
}
```

After

```php
class CategoryExtension extends CompilerExtension
{
  public function beforeCompile(): void
  {
    MappingHelper::of($this)
        ->addAnnotation( __DIR__ . '/../app/Model/Database', 'App\Model\Database')
        ->addAnnotation( __DIR__ . '/../../modules/Forum/Database', 'Forum\Modules\Database');
  }
}
```

### CODE REVIEW

**Comment by f3l1x** [comment]
Ref: https://github.com/Archette/Currency/blob/dfe3e6d0967018eebf9311621ee324906300e527/src/CurrencyExtension.php#L39

File: .docs/README.md [hunk 1]
```diff
@@ -235,31 +272,32 @@ extensions:
...
 
-You can use the predefined `TEntityMapping` trait in your compiler extensions. Be careful, you have to call it in `beforeCompile` phase.
+**MappingHelper**
+
+You can use the predefined `MappingHelper` helper class in your compiler extensions. Be careful, you have to call it in `beforeCompile` phase.
 
 ```php
-use Nette\DI\CompilerExtension;
-use Nettrine\ORM\DI\Traits\TEntityMapping;
+use Nette\DI\CompilerExtension;use Nettrine\ORM\DI\Helpers\MappingHelper;
```
Comment (adaamz) [inline .docs/README.md:289]:
```suggestion
use Nette\DI\CompilerExtension;
use Nettrine\ORM\DI\Helpers\MappingHelper;
```

**Comment by petrparolek** [comment]
thanks fo fix in b0e66d6f78

File: .docs/README.md (hunk 1 above)
Comment (f3l1x) [reply .docs/README.md]: Thx.

**Pull request merged**